dispatcher/backend/src/routes/utils.py

```diff
@@ -97,14 +97,16 @@ def remove_s3_secrets(response: dict):
         else:
             if not isinstance(response[key], str) or "://" not in response[key]:
                 continue
-            url = urlparse(response[key])
-            response[key] = url._replace(
-                query="&".join(
-                    [
-                        f"{key}={value}"
-                        for key, values in parse_qs(url.query).items()
-                        if str(key).lower() not in ["keyid", "secretaccesskey"]
-                        for value in values
-                    ]
-                )
-            ).geturl()
+            for part in [part for part in response[key].split() if "://" in part]:
```

Comment:
What's this supposed to be? Why splitting on space? Why not mame it `urlpart` to make it clearer? This will be hard to maintain

Reply:
The `split` is splitting on space and carriage returns and tabs.

I nevertheless agree that naming is bad. I would suggest to use this:

```
for urlpart in [word for word in response[key].split() if "://" in word]:
```

WDYT?

Reply:
Yes that's better. What source data could trigger this?

Reply:
This is typically useful for all logs, which might contain secrets if some scraper is not redacting them properly. We could even decide to not redact them anymore in the scraper so that Zimfarm admins have access to the plain text of the secret.